src/App.js

```diff
@@ -1,22 +1,57 @@
+import { useState, useEffect } from 'react';
+
 import './App.css';
 
 import ApplicatioHeader from './components/Layout/ApplicationHeader';
 import ApplicationBody from './components/Layout/ApplicationBody';
 import ApplicationFooter from './components/Layout/ApplicationFooter';
 
+import Modal from './components/Modal/Modal';
+import ModalContext from './contexts/modal-context';
 
-const App = () => (
-  <div className="app-layout">
-    <div className="app-layout-header">
-      <ApplicatioHeader titleLabel={'ReactJS Course'}></ApplicatioHeader>
-    </div>
-    <div className="app-layout-body">
-      <ApplicationBody></ApplicationBody>
-    </div>
-    <div className="app-layout-footer">
-      <ApplicationFooter></ApplicationFooter>
+
+const App = () => {
+  useEffect(() => { document.title = 'Mikalai Marozau | ReactJS Course' }, []);
+
+  // modal handlers
+  const [isModalActive, setIsModalActive] = useState(false);
+
+  const createModalHandler = (type, title, message, onClose, onConfirm) => {
+    setModalData({ type: type, title: title, message: message, onClose: onClose, onConfirm: onConfirm });
+    setIsModalActive(true);
+  };
+
+  const closeModalHandler = (result) => {
+    if (!result && modalData.onClose) {
+      modalData.onClose();
+    } else if (result && modalData.onConfirm) {
+      modalData.onConfirm();
+    }
+
+    setIsModalActive(false);
+    setModalData({ createModal: createModalHandler });
+  };
+
+  const [modalData, setModalData] = useState({ createModal: createModalHandler });
```

Comment:
Can you please place definition of states on the top of components code?

Reply:
Hm. It uses createModalHandler() in initial object properties. But createModalHandler() itself uses setModalData(). If i'll just move it to the top it will not recognize createModalHandler(), if i declare createModalHandler() using function - then it'd have stale closure. If there is some pattern/code method to solve it in best way?

Reply:
Moved to ModalManager wrapper component.